fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java

```diff
@@ -64,11 +66,46 @@ public DerbyMaster(String database) {
         }
 
         // Derby Server Properties are now set using the System Stored procedures.
-        try {
-            DerbyServerPropertiesMgr.setServerProperties(DEBUG, getConnection());
+        try (Connection c = getConnection()) {
+            DerbyServerPropertiesMgr.setServerProperties(DEBUG, c);
+            c.commit();
+        } catch (SQLException e) {
+            logger.log(Level.WARNING, "Derby Server Properties not set", e);
+        }
+    }
+    
+    /**
+     * Derby setSchema fails if the schema doesn't exist, so we need to create that
+     * now in order for our connections to succeed when we build out the FHIR database
+     * @param schemaName
+     */
+    public void createSchemaIfNeeded(String schemaName) {
+        DataDefinitionUtil.assertSecure(schemaName);
+        boolean createSchema = false;
+        try (Connection c = getConnection()) {
+            try {
+                c.setSchema(schemaName);
+            } catch (SQLException x) {
+                // schema doesn't exist, so we need to create it
+                createSchema = true;
+            }
+
+            if (createSchema) {
+                final String createSchemaDDL = "CREATE SCHEMA " + schemaName;
+                JdbcTarget target = new JdbcTarget(c);
+                target.runStatement(DERBY_TRANSLATOR, createSchemaDDL);
+                
+                try {
+                    c.commit();
+                }
+                catch (SQLException x) {
```

Comment:
```suggestion
                } catch (SQLException x) {
```
This is just messy to look at

Reply:
What is?